build212.toml

```diff
@@ -18,29 +18,29 @@ testFrameworks = [
 ]
 
 [module.pine]
-root    = "shared"
-sources = ["shared/src/main/scala"]
+# root    = "."
+sources = ["src/main/scala"]
 targets = ["js", "jvm"]
 
 [module.pine.test]
-sources   = ["shared/src/test/scala"]
+sources   = ["src/test/scala"]
 targets   = ["js", "jvm"]
 scalaDeps = [
-  ["org.scalatest" , "scalatest" , "3.0.7" ],
+  ["org.scalatest" , "scalatest" , "3.0.8" ],
   ["org.scalacheck", "scalacheck", "1.14.0"]
 ]
 
 [module.pine.jvm]
-root    = "jvm"
-sources = ["jvm/src/main/scala"]
+# root    = "jvm"
+sources = ["src/main/scala-jvm"]
 
 [module.pine.js]
-root    = "js"
-sources = ["js/src/main/scala"]
+# root    = "js"
+sources = ["src/main/scala-js"]
```

Comment:
Would it be possible to retain the original folder structure? Otherwise, we cannot generate an IntelliJ project.

Reply:
That might be tough...

I had to use sbt-projectmatrix to allow me to cross-build across different organisations, and it has some strong opinions about where source files should live, following much the same pattern seen in cats and shapeless.

Obviously IntelliJ will add support for projectmatrix if it becomes more popular (I suspect it will, it's a much cleaner approach than most projects are having to use for crossbuilding right now, just take a look at the build file for cats!).

I didn't rank this as too important.  After all, pine was going very happily for a while without literal type support in IntelliJ, it's what led me on a journey to discover bloop and metals :)

If it _is_ a deal-breaker though, I can investigate something more ingenious in the meantime. 
 Using `unmanagedSourceDirectories` to add the original directories to the matrix _might_ work, though I'm far from certain here.

Reply:
FYI: https://gitter.im/JetBrains/intellij-scala?at=5d1893eaddd2c007c7339da7

Reply:
At the moment, IntelliJ imposes two limitations on the directory layout of modules, particularly:

- The root path of a module cannot overlap with another module's root path
- A module's source and test files must be in folders underneath the configured root path

I checked sbt-projectmatrix's sources and it does provide any possibility to customise the platform paths.

At least for Seed we can partially fix the IDEA project generation with the following settings:

```
[module.pine]
root = "src/main/scala"
...
 
[module.pine.jvm]
root = "src/main/scala-jvm"
...
 
[module.pine.js]
root = "src/main/scala-js"
...
```

This satisfies the first requirement from above, but the test folder is not underneath the root path anymore. The project generation still works as expected and IntelliJ will now hide the tests.

I am fine with the new file structure given that developers will not notice anything unless they want to write test cases. Hopefully, IntelliJ's module restrictions will be lifted in the future.

Reply:
Also, we need to make the same changes in `build211.toml`. Afterwards, the Drone build should pass again.

Reply:
That root path is an even bigger problem with IntelliJ.  Because:

- projectmatrix spits out a project for every scala version
- intelliJ imports every project as a module
- the IDE only allows source folders beneath the project root

You end up in a position where you have projects/modules that really _need_ the same root because they're working from the exact same source files... it's only the compiler and maybe a handful of dependencies that differs.

Reply:
In that case, Pine developers can use Seed to generate the IntelliJ project until sbt-projectmatrix builds can be imported directly.

Reply:
Yup - I've been getting up to speed on seed, it looks very interesting.

Going to have to figure a way to de-dupe those toml files across different scala versions though!

Reply:
Agreed. We could use Seed's inheritance feature. For example, create a `base.toml` file that defines all the modules. Then, `build212.toml` includes it and overrides the Scala version and compiler flags.

Reply:
I'm actually taking a slightly different tack - seeing if I can generate the seed config via an SBT plugin.  That way **everything** can be kept in sync.  But that's not a pre-requisite for this PR

Reply:
I was actually thinking of adding support for generating sbt build files to Seed. This makes sense since Seed already positions itself more as a generation tool than a full-fledged build tool. The only remaining reason for me to use sbt is for publishing artefacts, but this will also be eventually implemented.